World Events and Global Events Unpicked

Shnupbups opened the pull request.

Shnupbups committed:
World Events and Global Events Unpicked

Shnupbups committed:
Update world_events.unpick

Earthcomputer commented on unpick-definitions/global_events.unpick:
```diff
@@ -0,0 +1,10 @@
+constant global_events net/fabricmc/yarn/constants/GlobalEvents ENDER_DRAGON_DIES
+constant global_events net/fabricmc/yarn/constants/GlobalEvents END_PORTAL_OPENED
+
+target_method net/minecraft/world/World syncGlobalEvent (ILnet/minecraft/util/math/BlockPos;I)V
+	param 0 global_events
+target_method net/minecraft/client/render/WorldRenderer processGlobalEvent (ILnet/minecraft/util/math/BlockPos;I)V
+	param 0 global_events
```
You sure there aren't more than this (and for the other one too)?

Shnupbups replied:
Pretty sure. These are all the ones handled by WorldRenderer at least...

Earthcomputer replied:
I was thinking maybe in the packet.

liach replied:
Imo if we have missing ones we can always add them later. Indeed, the best place to check is in the packet, as there may be events that only have sound effects without render effects, for instance.

Shnupbups replied:
Sound effect events are handled in WorldRenderer too
(which kinda makes the WorldRenderer name a bit inaccurate but whatever)

liach reviewed (changes requested):
The javadocs should use `.` or start new paragraphs than simple line breaks. Javadoc tool interpret line breaks as spaces.

Shnupbups committed:
stuff

consolidated WorldEvents and GlobalEvents into one
renamed CROPS_FERTILIZED to PLANT_FERTILIZED
added packet stuff to target methods
added punctuation to javadoc

i509VCB commented on src/constants/java/net/fabricmc/yarn/constants/WorldEvents.java:
```diff
@@ -0,0 +1,434 @@
+package net.fabricmc.yarn.constants;
+
+public final class WorldEvents {
```
This really needs a private constructor to prevent instantiation.

Shnupbups committed:
whoops

Shnupbups committed:
Update WorldEvents.java

OroArmor commented:
I approve. This is way better than #1652

YanisBft commented:
Missing methods I've noticed:
- DoorBlock.getOpenSoundEventId()
- DoorBlock.getCloseSoundEventId()

Shnupbups committed:
door stuff

Shnupbups commented:
Added the door ones but no idea how I'd do the potion one

Earthcomputer commented:
The magic constants in `PotionEntity.onCollision` should already be covered by `World.syncWorldEvent`, the latest unpick should be smart enough to detect those constants.

YanisBft commented:
Do the targeted methods with "return" (the DoorBlock ones) work for anyone? Because they do not for me

Earthcomputer commented:
Works fine for me
![Screenshot from 2021-03-28 14-02-10](https://user-images.githubusercontent.com/13084089/112753134-3cbd3580-8fce-11eb-8484-7908a9cbc2a4.png)

YanisBft reviewed (approved):
The PotionEntity doesn't work for me, but the others are fine

liach commented on src/constants/java/net/fabricmc/yarn/constants/WorldEvents.java:
```diff
@@ -0,0 +1,437 @@
+package net.fabricmc.yarn.constants;
+
+public final class WorldEvents {
```
I hope you can add javadoc links to methods that actually process world events, like `net.minecraft.world.WorldAccess#syncWorldEvent(net.minecraft.entity.player.PlayerEntity, int, net.minecraft.util.math.BlockPos, int)`

Shnupbups committed:
Merge branch '21w11a' into 21w11a-eventsunpicked

Shnupbups committed:
Improved Javadoc

Added doc for Constant class itself, added linebreaks, and info on everywhere each event is called

Shnupbups committed:
Update WorldEvents.java

Shnupbups commented:
javadoc go brrr

liach reviewed (approved):
Thank you very much, very high quality!

Juuxel commented on src/constants/java/net/fabricmc/yarn/constants/WorldEvents.java:
```diff
@@ -0,0 +1,558 @@
+package net.fabricmc.yarn.constants;
+
+/**
+ * Constants of World Event IDs.<br>
+ * World Events are used to trigger things on the client from the server side.
+ * Most commonly, playing sound events or spawning particles.<br>
+ * Some events have an extra data integer sent alongside them.<br>
+ * Some events are global, meaning they will be sent to every player regardless of their position.<br>
+ * Events are sent from the server to the client using {@link net.minecraft.network.packet.s2c.play.WorldEventS2CPacket WorldEventS2CPacket},
```
`<p>` would be better than `<br>` here. (Also possibly when separating the 2nd description line of the individual constants from the "called by" sections)

Shnupbups committed:
changed some <br>s to <p>s

liach commented:
Mind use an formatter like intellij so we have `<p>` tags on the beginning of lines than the end of the lines?

Shnupbups committed:
Merge branch '21w11a' into 21w11a-eventsunpicked

Shnupbups committed:
javadoc tweak

moved <p> and <br> tags to start of new lines as opposed to end of previous lines
tweaked wording of MUSIC_DISC_PLAYED
qualified mentions of java.util.List and java.util.Random
made some wording more consistent

liach commented:
Just fyi, the build failure is because the gradle wrapper validator was offline.

modmuss50 commented:
A rebuild did the trick 👍
